src/kvstore/kvstore_dist_server.h

```diff
@@ -170,6 +187,33 @@ class KVStoreDistServer {
     app->Response(recved);
   }
 
+  void SetProfilerConfig(std::string params_str) {
+    std::vector<std::string> elems;
+    mxnet::kvstore::split(params_str, ',', std::back_inserter(elems));
+    std::vector<const char*> ckeys;
+    std::vector<const char*> cvals;
+    ckeys.reserve(elems.size());
+    cvals.reserve(elems.size());
+
+    for (int i=0; i < elems.size(); i++) {
+      std::vector<std::string> parts;
```

Comment:
you never push into the vector, right?  then [0] and [1] indexing is invalid addressing

Reply:
Elements are pushed using `std::back_inserter` in `split` method

Reply:
Elements are pushed using `std::back_insert` in `split` method

Reply:
oh ok missed that my mistake